nifi-nar-bundles/nifi-websocket-bundle/nifi-websocket-services-jetty/src/main/java/org/apache/nifi/websocket/jetty/JettyWebSocketServer.java

```diff
@@ -0,0 +1,258 @@
+            final URI requestURI = servletUpgradeRequest.getRequestURI();
+            final int port = requestURI.getPort();
+            final JettyWebSocketServer service = portToControllerService.get(port);
+
+            if (service == null) {
+                throw new RuntimeException("No controller service is bound with port: " + port);
+            }
+
+            final String path = requestURI.getPath();
+            final WebSocketMessageRouter router;
+            try {
+                router = service.routers.getRouterOrFail(path);
+            } catch (WebSocketConfigurationException e) {
+                throw new IllegalStateException("Failed to get router due to: "  + e, e);
+            }
+
+            final RoutingWebSocketListener listener = new RoutingWebSocketListener(router) {
+                @Override
+                public void onWebSocketConnect(Session session) {
+                    final WebSocketPolicy currentPolicy = session.getPolicy();
+                    currentPolicy.setInputBufferSize(service.configuredPolicy.getInputBufferSize());
+                    currentPolicy.setMaxTextMessageSize(service.configuredPolicy.getMaxTextMessageSize());
+                    currentPolicy.setMaxBinaryMessageSize(service.configuredPolicy.getMaxBinaryMessageSize());
+                    super.onWebSocketConnect(session);
+                }
+            };
+
+            return listener;
+        }
+    }
+
+    @OnEnabled
+    @Override
+    public void startServer(final ConfigurationContext context) throws Exception {
+
+        if (server != null && server.isRunning()) {
+            getLogger().info("A WebSocket server is already running. {}", new Object[]{server});
+            return;
+        }
+
+        configuredPolicy = WebSocketPolicy.newServerPolicy();
+        configurePolicy(context, configuredPolicy);
+
+        server = new Server();
+
+        final ContextHandlerCollection handlerCollection = new ContextHandlerCollection();
+
+        final ServletContextHandler contextHandler = new ServletContextHandler();
+        servletHandler = new ServletHandler();
+        contextHandler.insertHandler(servletHandler);
+
+        handlerCollection.setHandlers(new Handler[]{contextHandler});
+
+        server.setHandler(handlerCollection);
+
+        listenPort = context.getProperty(LISTEN_PORT).asInteger();
+        final SslContextFactory sslContextFactory = createSslFactory(context);
+
+        final ServerConnector serverConnector = createConnector(sslContextFactory, listenPort);
+
+        server.setConnectors(new Connector[] {serverConnector});
+
+        servletHandler.addServletWithMapping(JettyWebSocketServlet.class, "/*");
+
+        // Need to specify classloader, otherwise since the callstack doesn't have any nifi specific class, so it can't use nar.
+        try (NarCloseable closeable = NarCloseable.withComponentNarLoader(WebSocketServerFactory.class)) {
```

Comment:
Thanks Oleg, I removed the use of NarCloseable and confirmed that JettyWebSocketServer works without this now.